security/backends/common/is_core.py

```diff
@@ -150,7 +163,13 @@ def error_message_code(self, obj):
     @short_description(_('source'))
     def display_source(self, obj, request):
         parent_log_obj = self._get_parent_log_obj(obj)
-        return render_model_object_with_link(request, parent_log_obj) if parent_log_obj else None
+        if not parent_log_obj:
+            return None
+        core_parent_model = get_model_core(parent_log_obj.__class__)
+        verbose_name_with_last_id_digits = '{} {}'.format(
+            core_parent_model.verbose_name, core_parent_model.short_id(parent_log_obj)
+        )
+        return render_model_object_with_link(request, parent_log_obj, display_value=verbose_name_with_last_id_digits)
```

Comment:
tak jsem o tom přemýšlel a to display_value není dobrý nápad, ty budeš muset opravdu přepsat metodu str, jelikož pokud je log zobrazen z nějakého jiného modelu tak se právě metoda str používá. Víš co, udělej to verbose_name_with_last_id_digits přímo do str funkce a zde to odstraň, tím pádem se bude všude používat standardní str.

jediný problém budeš mít s verbose name. Ale normálně natvrďáka do každého logu přidej str metodu a ten verbose tam dej natvrdo jako gettext string

Reply:
ok, I have changed that and implemented __str__ method for each model.